frontends/concrete-python/concrete/fhe/mlir/operations/min_max.py

```diff
@@ -0,0 +1,210 @@
+        ctx.error(highlights)
+
+    if x.is_scalar:
+        return x
+
+    if axes is None:
+        axes = []
+
+    axes = list(
+        set([int(axes)] if isinstance(axes, (int, np.integer)) else [int(axis) for axis in axes])
+    )
+
+    input_dimensions = len(x.shape)
+    for i, axis in enumerate(axes):
+        if axis < 0:
+            axes[i] += input_dimensions
+
+    if len(axes) == 0 or len(axes) == len(x.shape):
+        x = ctx.flatten(x)
+
+    resulting_element_type = ctx.element_typeof(resulting_type)
+    if len(x.shape) == 1:
+        result = reduce(ctx, resulting_element_type, x, operation=operation)
+        if keep_dims:
+            result = ctx.reshape(result, shape=resulting_type.shape)
+        return result
+
+    class Mock:
+        indices: Set[Tuple[int, ...]]
+
+        def __init__(self, index: Tuple[int, ...]):
+            self.indices = {index}
+
+        def __repr__(self) -> str:
+            return f"{self.indices}"
+
+        def combine(self, other: "Mock") -> "Mock":
+            result = deepcopy(self)
+            for index in other.indices:
+                result.indices.add(index)
+            return result
+
+    mock_input = []
+    for index in np.ndindex(x.shape):
+        mock_input.append(Mock(index))
+    mock_input = np.array(mock_input).reshape(x.shape)
+
+    def accumulate(mock1, mock2):
+        return mock1.combine(mock2)
+
+    mock_output = np.frompyfunc(accumulate, 2, 1).reduce(
+        mock_input,
+        axis=tuple(axes),
+        keepdims=keep_dims,
+    )
+
+    sample_mock = mock_output.flat[0]
+
+    sample_mock_indices = sample_mock.indices
+    sample_mock_index = next(iter(sample_mock_indices))
+
+    number_of_comparisons = len(sample_mock_indices)
+    number_of_indices = len(sample_mock_index)
+    index_shape = resulting_type.shape
+
+    to_compare = []
+    for _ in range(number_of_comparisons):
+        indices = []
+        for _ in range(number_of_indices):
+            index = np.zeros(index_shape, dtype=np.int64)  # type: ignore
+            indices.append(index)
+        to_compare.append(tuple(indices))
+
+    for position in np.ndindex(mock_output.shape):
+        mock_indices = list(mock_output[position].indices)
+        for i in range(number_of_comparisons):
+            for j in range(number_of_indices):
+                to_compare[i][j][position] = mock_indices[i][j]  # type: ignore
+
+    extracted = []
+    extracted_type = ctx.tensor(ctx.element_typeof(x), shape=resulting_type.shape)
+
+    for index in to_compare:  # type: ignore
+        extracted.append(ctx.index(extracted_type, x, index))
+
+    while len(extracted) > 1:
+        a = extracted.pop()
+        b = extracted.pop()
+
+        if operation == "min":
+            c = ctx.minimum(resulting_type, a, b)
+        else:
+            c = ctx.maximum(resulting_type, a, b)
+
+        c.set_original_bit_width(x.original_bit_width)
+        extracted.insert(0, c)
+
+    return extracted[0]
```

Comment:
I (for one) am not able to to understand all that without debugging throught it. Could you provide some explanations ?

Reply:
Actually I don't think anything gets tested past line 82, see my comment below.

Reply:
I've replied below.

Reply:
Of course, I'll add comments!